agent/structs/structs.go

```diff
@@ -1297,6 +1300,83 @@ type ServiceAddress struct {
 	Port    int
 }
 
+type ServicePort struct {
+	Name    string
+	Port    int
+	Default bool
+}
+
+type ServicePorts []ServicePort
+
+func (sp ServicePorts) Validate() error {
```

Comment:
Avoiding cloning

```suggestion
func (sp *ServicePorts) Validate() error {
```

Reply:
@srahul3 ServicePorts is just a slice of ServicePort which is already a pointer and won't be copied.